src/audio/module_adapter/module/volume/volume_hifi3.c

```diff
@@ -91,6 +92,9 @@ static void vol_s24_to_s24_s32(struct comp_dev *dev, struct audio_stream *sink,
 	AE_SETCBEGIN0(buf);
 	AE_SETCEND0(buf_end);
 
+	bsource->consumed += samples << 2;
+	bsink->size += samples << 2;
+
 	while (samples) {
 		m = VOL_BYTES_TO_S32_SAMPLES(audio_stream_bytes_without_wrap(source, in));
```

Comment:
Same for hifi3 versions. No circular source/sink buffers wrap need if buffers are linear.

Reply:
Same, buffers are still circular -> resolve.